staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/applyconfiguration.go

```diff
@@ -306,3 +316,34 @@ func $.ApplyConfig.Type|public$() *$.ApplyConfig.ApplyConfiguration|public$ {
   return &$.ApplyConfig.ApplyConfiguration|public${}
 }
 `
+
+func (g *applyConfigurationGenerator) generateClientgenExtract(sw *generator.SnippetWriter, typeParams TypeParams) {
+	sw.Do(`
+// Extract$.ApplyConfig.Type|public$ extracts the applied configuration owned by fieldManager from
+// $.Struct|private$. If no managedFields are found in $.Struct|private$ for fieldManager, nil is returned.
```

Comment:
Out of curiosity, why nil instead of an empty applyconfig? Calling this functions probably means the caller has the intent of creating an applyconfig regardless of the managedfields. Wouldn't it be easier to just create it here?

Reply:
I see your point. I was thinking more about giving a signal to users to let them know that no managed fields were found for the field manager they provided.
 
Unfortunately, it not possible to distinguish between (1) didn't find any managed fields because the given fieldManager has never applied to the object and (2) didn't find any managed fields because all the fields owned by the field manager have been claimed by other managers.

If the user did manage field before, it's more likely they want to create the object regardless of what's happened to their field ownership since they last applied.

I suppose that if we don't provide a signal to indicate if the managed fields were found or not, they could always check the object directly to get the answer.

Reply:
I've updated to run an empty object and clarified exactly what is returned and under what conditions it can be returned (because it didn't seem 100% obvious to me that developers would know this).